fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SizeStatisticsCollector.java

```diff
@@ -52,26 +53,16 @@
 @API(API.Status.EXPERIMENTAL)
 public class SizeStatisticsCollector {
     @Nonnull
-    private final SubspaceProvider subspaceProvider;
-    private long keyCount;
-    private long keySize;
-    private long maxKeySize;
-    private long valueSize;
-    private long maxValueSize;
-    @Nonnull
-    private long[] sizeBuckets;
+    private SubspaceProvider subspaceProvider;
     @Nonnull
+    private SizeStatisticsCollectorCursor.SizeStatisticsResults sizeStatsResults;
+    @Nullable
     private RecordCursorContinuation continuation;
 
     private SizeStatisticsCollector(@Nonnull SubspaceProvider subspaceProvider) {
         this.subspaceProvider = subspaceProvider;
-        this.keyCount = 0;
-        this.keySize = 0;
-        this.maxKeySize = 0;
-        this.valueSize = 0;
-        this.maxValueSize = 0;
-        this.sizeBuckets = new long[Integer.SIZE];
         this.continuation = RecordCursorStartContinuation.START;
+        this.sizeStatsResults = getSizeStatisticsResultsInitialState();
```

Comment:
Hm, I think the static import here is a little confusing.

Is this not just `new SizeStatisticsCursor.SizeStatitisticsResults()`?

Reply:
If we made the constructor public then yes. See above comment on the design of this class.